code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -10,44 +10,101 @@ CHEAP LIGHTERS
 ZIPPO
 
 CIGARETTE PACKETS ARE IN FANCY.DM
+MATCHBOXES ARE ALSO IN FANCY.DM
 */
 
 ///////////
 //MATCHES//
 ///////////
+
 /obj/item/weapon/match
 	name = "match"
-	desc = "A simple match stick, used for lighting fine smokables."
+	desc = "A budget match stick, used to start fires easily, preferably at the end of a smoke."
 	icon = 'icons/obj/cigarettes.dmi'
-	icon_state = "match_unlit"
+	icon_state = "match"
+	item_state = "cig"
 	var/lit = 0
-	var/smoketime = 5
+	var/smoketime = 10
+	var/brightness_on = 1 //Barely enough to see where you're standing, it's a shitty discount match
 	heat_production = 1000
 	w_class = 1.0
 	origin_tech = "materials=1"
 	attack_verb = list("burnt", "singed")
 
+/obj/item/weapon/match/New()
+
+	..()
+	update_brightness() //Useful if you want to spawn burnt matches, or burning ones you maniac
+
+/obj/item/weapon/match/examine(mob/user)
+
+	..()
+	switch(lit)
+		if(1)
+			user << "The match is lit."
+		if(0)
+			user << "The match is unlit and ready to be used."
+		if(-1)
+			user << "The match is burnt."
+
+//Also updates the name, the damage and item_state for good measure
+/obj/item/weapon/match/update_icon()
+
+	switch(lit)
+		if(1)
+			name = "lit [initial(name)]"
+			item_state = "[initial(item_state)]on"
+			icon_state = "[initial(icon_state)]_lit"
+			damtype = "fire"

```

Comment:
This seems correct from what a quick look-up attests, are you sure you aren't mixing it up with another variable ?